### PR TITLE
UCT/TCP: Fix iface flush

### LINKED ISSUE #2089: Failure in tcp/test_ucp_perf.envelope/0
Happened with valgrind:

```
12:30:45 [----------] 1 test from tcp/test_ucp_perf
12:30:45 [ RUN      ] tcp/test_ucp_perf.envelope/0
12:30:45 [     INFO ]                tag latency : 385.990 usec
12:30:46 [     INFO ]            tag iov latency : 252.335 usec
12:30:46 [     INFO ]                     tag mr : 0.002 Mpps
12:30:47 [     INFO ]                tag sync mr : 0.003 Mpps
12:30:47 [     INFO ]                tag wild mr : 0.003 Mpps
12:30:48 [hpc-test-node3:32079:0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x6364)
```

Trace:
```
#19 <signal handler called>
#20 ucs_queue_push (queue=0x435c5f8, elem=0x4072fe8) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucs/datastruct/queue.h:56
#21 ucs_rcache_unmapped_callback (event_type=<optimized out>, event=<optimized out>, arg=0x435c560)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucs/sys/rcache.c:275
#22 0x000000000544ff0e in ucm_event_dispatch (event=0xf3777c0, event_type=UCM_EVENT_VM_UNMAPPED)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucm/event/event.c:123
#23 ucm_dispatch_vm_munmap (length=12288, addr=0x4119000) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucm/event/event.c:171
#24 ucm_munmap (addr=0x4119000, length=12288) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucm/event/event.c:210
#25 0x00000000094d2398 in mlx5_free_actual_buf () from /usr/lib64/mlnx_ofed/valgrind/libmlx5-rdmav2.so
#26 0x000000000950505c in mlx5_destroy_qp () from /usr/lib64/mlnx_ofed/valgrind/libmlx5-rdmav2.so
#27 0x00000000051bcae6 in uct_ib_md_umr_qp_destroy (md=0x427df00, md=0x427df00)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/uct/ib/base/ib_md.c:297
#28 uct_ib_md_close (uct_md=0x427df00) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/uct/ib/base/ib_md.c:1278
#29 0x00000000056757a7 in ucp_fill_resources (config=0x4299c10, context=0x4282760)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucp/core/ucp_context.c:711
#30 ucp_init_version (api_major_version=api_major_version@entry=1, api_minor_version=api_minor_version@entry=3, params=params@entry=0xf377bb0, config=0x4299c10, 
    context_p=context_p@entry=0x4369780) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucp/core/ucp_context.c:942
#31 0x0000000000739fd5 in ucp_init (context_p=0x4369780, config=<optimized out>, params=0xf377bb0)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/ucp/api/ucp.h:1106
---Type <return> to continue, or q <return> to quit---
#32 ucp_perf_setup (perf=0x4365630, params=0xffeffe250) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/tools/perf/libperf.c:1232
#33 0x000000000073ce85 in ucx_perf_run (params=0xffeffe250, result=0x42750c8)
    at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../src/tools/perf/libperf.c:1319
#34 0x00000000004729e9 in test_perf::thread_func (arg=0xffeffe250) at /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node3/worker/1/contrib/../test/gtest/common/test_perf.cc:165
#35 0x0000000006f88e25 in start_thread () from /usr/lib64/libpthread.so.0
#36 0x00000000082fd34d in clone () from /usr/lib64/libc.so.6 
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5543/label=hpc-test-node3,worker=1/console

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2089

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5603/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5607/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5613/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5614/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5619/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich pls take a look

File: src/uct/tcp/tcp_iface.c [hunk 1]
```diff
@@ -155,6 +155,23 @@ unsigned uct_tcp_iface_progress(uct_iface_h tl_iface)
...
+                                        uct_completion_t *comp)
+{
+    uct_tcp_iface_t *iface = ucs_derived_of(tl_iface, uct_tcp_iface_t);
+
+    if (comp != NULL) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (iface->outstanding) {
+        UCT_TL_IFACE_STAT_FLUSH_WAIT(&iface->super);
```
Comment (brminich) [inline src/uct/tcp/tcp_iface.c]: looks like need to return `UCS_INPROGRESS` or `UCS_ERR_NO_RESOURCE` here.  Should not it be caught by UCT flush test?

Comment (yosefe) [reply src/uct/tcp/tcp_iface.c]: i introduced this bug when tried to fix coverity :( gtest catch it, but after a very large amount of iterations..

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5629/ for details (Mellanox internal link).

**Pull request merged**